src/fsharp/FSharp.Build/Microsoft.FSharp.Targets

```diff
@@ -400,6 +400,63 @@ this file.
 
     </Target>
 
+    <!--
+
+    The old reference assemblies are no longer installed to `Program Files (x86)\Reference Assemblies\Microsoft\FSharp`.
+
+    To avoid breaking legacy projects the hint path for FSharp.Core will be rewritten if and only if the existing hint
+    path is unchanged from the original legacy templates and if a new reference assembly can be found at the new location.
+
+    -->
+    <Target Name="RedirectFSharpCoreReferenceToNewRedistributableLocation" BeforeTargets="ResolveAssemblyReferences">
```

Comment:
I'm not sure how I feel about this target.  It's very clever, but may also be very surprising to people, and I feel like various bits of tooling that only use the evaluated MSBuild model will be wrong because they will still use the old hint path.

With that said, I'm not sure of anything better :-/

Reply:
Hmm. What did the old templates put in user proj files?

Does this do the right thing in design-time builds?

Reply:
I didn't see any issues with design time builds, but I'll do another pass checking for this specifically.

The old templates created the project files like this:

``` xml
<ItemGroup>
  ...
  <Reference Include="FSharp.Core">
    <Name>FSharp.Core</Name>
    <AssemblyName>FSharp.Core.dll</AssemblyName>
    <HintPath>$(MSBuildProgramFiles32)\Reference Assemblies\Microsoft\FSharp\.NETFramework\v4.0\$(TargetFSharpCoreVersion)\FSharp.Core.dll</HintPath>
  </Reference>
  ...
</ItemGroup>
```

and we want to make sure we don't break those scenarios.